lib/div-alg.gi

```diff
@@ -1631,6 +1631,10 @@ if p < q then
   q:=a;
 fi;
 
+if q<-1 or (q=-1 and not(p=-1 or IsPrimeInt(p))) or (q>-1 and not(IsPrimeInt(q))) or (q>-1 and IsPrimeInt(q) and not(IsPrimeInt(p))) then
```

Comment:
Would it be more readable to write the condition as

    not ForAll( [p,q], t -> t=-1 or IsPosInt(t) and IsPrimeInt(t) );

?